src/ESPEasyRules.ino

```diff
@@ -304,28 +304,84 @@ void replace_EventValueN_Argv(String& line, const String& argString, unsigned in
   }
 }
 
-void substitute_eventvalue(String& line, const String& event) {
-  if (line.indexOf(F("%eventvalue")) == -1) {
-    return; // Nothing to replace.
+void process_substring(String& line) {
+  // Syntax like 12345[substring:8:12:ANOTHER HELLO WORLD]67890
+  int startIndex;
+  while ((startIndex = line.indexOf(F("[substring:"))) != -1) {  
+    String substCmd = line.substring(startIndex, line.indexOf("]", startIndex) + 1);
+    int idxStart = substCmd.indexOf(":") + 1;
+    int idxEnd = substCmd.indexOf(":", idxStart);
+    String substStart = substCmd.substring(idxStart, idxEnd);
+    idxStart = idxEnd + 1;
+    idxEnd = substCmd.indexOf(":", idxStart);
+    String substEnd = substCmd.substring(idxStart , idxEnd);
+    idxStart = idxEnd + 1;
+    idxEnd = substCmd.indexOf("]", idxStart);
+    String stringValue = substCmd.substring(idxStart, idxEnd);
+  
+    line.replace(substCmd, stringValue.substring(substStart.toInt(), substEnd.toInt()));
   }
+}
 
-  if (event.charAt(0) == '!') {
-    line.replace(F("%eventvalue%"), event); // substitute %eventvalue% with
-                                            // literal event string if
-                                            // starting with '!'
-  } else {
-    int equalsPos = event.indexOf("=");
+void process_strtol(String& line) {
+  // Syntax like 1234[strtol:16:38]7890
+  int startIndex;
+  while ((startIndex = line.indexOf(F("[strtol:"))) != -1) {    
+    String strolCmd = line.substring(startIndex, line.indexOf("]", startIndex) + 1);
+    int idxStart = strolCmd.indexOf(":") + 1;
+    int idxEnd = strolCmd.indexOf(":", idxStart);
+    String base = strolCmd.substring(idxStart, idxEnd);
+    idxStart = idxEnd + 1;
+    idxEnd = strolCmd.indexOf("]", idxStart);
+    String stringValue = strolCmd.substring(idxStart, idxEnd);
+       
+    line.replace(strolCmd, String(strtol(stringValue.c_str(), NULL, base.toInt())));   
+  }
+}
```

Comment:
This function is almost the same as `process_substring` and `process_div100ths`.
Maybe you can make a function to determine the ranges?
Just to make sure we don't have a lot of code duplication.

Reply:
I joined the functions in https://github.com/crathje/ESPEasy/commit/d238593ac0b039cdc10a0b920de9a3fb20f128e7